src/main/java/de/dennisguse/opentracks/fragments/IntervalsRecordingFragment.java

```diff
@@ -0,0 +1,136 @@
+package de.dennisguse.opentracks.fragments;
+
+import android.os.Bundle;
+import android.os.Handler;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.ViewModelProvider;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.content.data.Track;
+import de.dennisguse.opentracks.util.PreferencesUtils;
+import de.dennisguse.opentracks.util.UnitConversions;
+import de.dennisguse.opentracks.viewmodels.IntervalStatistics;
+import de.dennisguse.opentracks.viewmodels.IntervalStatisticsModel;
+import de.dennisguse.opentracks.views.IntervalListView;
+import de.dennisguse.opentracks.views.IntervalReverseListView;
+
+/**
+ * A fragment to display the intervals from recording track.
+ */
+public class IntervalsRecordingFragment extends Fragment implements IntervalListView.IntervalListListener {
+
+    private static final String TAG = IntervalsRecordingFragment.class.getSimpleName();
+
+    private static final long UI_UPDATE_INTERVAL = 5 * UnitConversions.ONE_SECOND_MS;
+
+    private IntervalStatisticsModel viewModel;
+    private IntervalReverseListView intervalListView;
+
+    private class IntervalRunner implements Runnable {
+        private boolean stopped = false;
+
+        @Override
+        public void run() {
+            if (!stopped) {
```

Comment:
Why not check isResume() of parent class?
Then we don't need stop.
See StatisticsRecordingFragment:107.